File: euler/problem 3.py


def est_premier(n):
    for k in range(2,int(n**0.5)):
        if n%k==0:
            return False
    return True

def largest_prime_factor(n):
    m=[]
    for k in range(1,int(n**0.5)+1):
        if est_premier(k) and n%k==0:
            if est_premier(n/k):
                m.append(n/k)
            m.append(k)
            print(k)
    return max(m)

print(largest_prime_factor(600851475143))

File: problem 3.ml
let est_premier n =
  let rec aux n v =
    if n = 2 || v=1 then true
    else n mod v <> 0 && aux n (v-1) 
  in aux n (int_of_float (1.+.(float_of_int n)**0.5)) ;;

let largest_prime_factor n =
    let rec aux q v = 
      if q = 1 then v else
      if est_premier q && n mod q = 0 then aux (q-1) (max v (max (n/q) q))
      else aux (q-1) v 
    in aux (int_of_float (1.+.(float_of_int n)**0.5)) 0 ;;


(* nombre trop grand ptdrrrr*)